src/libcoretest/option.rs

```diff
@@ -269,4 +271,10 @@ fn test_cloned() {
     assert_eq!(opt_ref_ref.clone(), Some(&val_ref));
     assert_eq!(opt_ref_ref.clone().cloned(), Some(&val));
     assert_eq!(opt_ref_ref.cloned().cloned(), Some(1));
+
+    // Mutable ref works
+    assert_eq!(opt_mut.as_mut().cloned(), Some(1));
+
+    // Deep Deref works
+    assert_eq!(opt_box.cloned(), None);
```

Comment:
Do I understand this correctly and the result of `opt_box.cloned()` is an `Option<u32>`?